src/overmap_ui.cpp

```diff
@@ -801,17 +807,67 @@ tripoint display( const tripoint &orig, const draw_data_t &data = draw_data_t()
             std::string color_notes = _( "Color codes: " );
             for( auto color_pair : get_note_color_names() ) {
                 // The color index is not translatable, but the name is.
-                color_notes += string_format( "%s:%s, ", color_pair.first.c_str(),
-                                              _( color_pair.second.c_str() ) );
+                color_notes += string_format( "%1$s:<color_%3$s>%2$s</color>, ", color_pair.first.c_str(),
+                                              _( color_pair.second.c_str() ), string_replace( color_pair.second, " ", "_" ).c_str() );
             }
-            color_notes = color_notes.replace( color_notes.end() - 2, color_notes.end(), "." );
+
+            std::string helper_text = string_format( ".\r\n \n%s\r\n%s\r\n%s\r\n ",
+                                      _( "Type GLYPH:TEXT to set a custom glyph." ),
+                                      _( "Type COLOR;TEXT to set a custom color." ),
+                                      _( "Examples: B:Base | g;Loot | !:R;Minefield" ) );
+            color_notes = color_notes.replace( color_notes.end() - 2, color_notes.end(), helper_text );
+            std::string title = _( "Note:" );
+
             const std::string old_note = overmap_buffer.note( curs );
-            const std::string new_note = string_input_popup()
-                                         .title( _( "Note (X:TEXT for custom symbol, G; for color):" ) )
-                                         .width( 45 )
-                                         .text( old_note )
-                                         .description( color_notes )
-                                         .query_string();
+            std::string new_note = old_note, tmp_note;
+
+            bool done = false, esc_pressed = false;
+            do {
+                // Popup must be created anew because the description is only
+                // printed when the window is created. This only happens once
+                // every keystroke, however.
+                string_input_popup &input_popup = string_input_popup();
```

Comment:
This won't work:

![image](https://user-images.githubusercontent.com/16213433/45295595-e9fef980-b507-11e8-96d4-dc68f1561673.png)

Reply:
I get a similar error if I don't use a reference, something about trying to use a deleted constructor. 
`Error	C2280	'string_input_popup::string_input_popup(const string_input_popup &)': attempting to reference a deleted function	Cataclysm	G:\Projects\Cataclysm\Cataclysm-DDA\src\overmap_ui.cpp	823`

I'm building with VS by the way. I've noticed that other parts of the code use the copy constructor, but I can't wrap my head around why it's not working as intended.

Reply:
Yeah, VS doesn't complain about that

Anyway - see https://github.com/CleverRaven/Cataclysm-DDA/pull/25418#pullrequestreview-153743360 - it makes both CLANG, GC++ and VS happy.